src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -299,12 +299,20 @@ private InstanceCollection getTenantSpecificSourceCollection(String tenantId, Ma
   }
 
   private List<KafkaHeader> convertKafkaHeadersMap(Map<String, String> kafkaHeaders) {
-    return new ArrayList<>(List.of(
-      KafkaHeader.header(OKAPI_URL_HEADER, kafkaHeaders.get(OKAPI_URL_HEADER)),
-      KafkaHeader.header(OKAPI_TENANT_HEADER, kafkaHeaders.get(OKAPI_TENANT_HEADER)),
-      KafkaHeader.header(OKAPI_TOKEN_HEADER, kafkaHeaders.get(OKAPI_TOKEN_HEADER)),
-      KafkaHeader.header(OKAPI_USER_ID, kafkaHeaders.get(OKAPI_USER_ID)),
-      KafkaHeader.header(OKAPI_REQUEST_ID, kafkaHeaders.get(OKAPI_REQUEST_ID)))
-    );
+
+    var headers = (kafkaHeaders.get(OKAPI_REQUEST_ID) != null) ?
```

Comment:
`List<KafkaHeader> headers = new ArrayList<>(List.of(
    KafkaHeader.header(OKAPI_URL_HEADER, kafkaHeaders.get(OKAPI_URL_HEADER)),
    KafkaHeader.header(OKAPI_TENANT_HEADER, kafkaHeaders.get(OKAPI_TENANT_HEADER)),
    KafkaHeader.header(OKAPI_TOKEN_HEADER, kafkaHeaders.get(OKAPI_TOKEN_HEADER)),
    KafkaHeader.header(OKAPI_USER_ID, kafkaHeaders.get(OKAPI_USER_ID))
));

if (kafkaHeaders.get(OKAPI_REQUEST_ID) != null) {
  headers.add(KafkaHeader.header(OKAPI_REQUEST_ID, kafkaHeaders.get(OKAPI_REQUEST_ID)));
}

return headers;`